external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -47,21 +47,20 @@ def __post_init__(self):
             raise TypeError("times must be one of 'window' or 'start'")
 
     def get_record(self, name: str, ds: xr.Dataset, unstacked_dims: Sequence[str]):
-        for dim in unstacked_dims[:-1]:
-            if dim not in ds[name].dims:
-                raise ValueError("variable {} has no dimension {}".format(name, dim))
         if self.times == "start":
             ds = ds.isel(time=0)
-        ds = stack(ds, unstacked_dims)
+        # must subset dataset to one variable to avoid attaching dimensions
```

Comment:
Couldn't we just make `stack` operate on data arrays instead and not have to worry about this?

Reply:
We probably could. It may even "just work" - I'll try.

Reply:
It needed some other changes, but they're for the better. Updated.